app/src/main/java/org/wikipedia/ActivityLifecycleHandler.kt

```diff
@@ -14,13 +15,19 @@ class ActivityLifecycleHandler : ActivityLifecycleCallbacks, ComponentCallbacks2
 
     private var haveMainActivity = false
     var isAnyActivityResumed = false
+    private var currentActivity: Activity? = null
 
     fun haveMainActivity(): Boolean {
         return haveMainActivity
     }
 
+    fun isOnThisDayGameActivityVisible(): Boolean {
```

Comment:
This feels too "specific" to put into something as general as ActivityLifecycleHandler. It would be OK to expose a function like `getResumedActivity()` here, and call it from whatever consumer needs to check it. (Ideally nothing else should "know about" `OnThisDayGame...` classes except things that directly need to.)